utils/lldp.go

```diff
@@ -0,0 +1,186 @@
+/***
+Copyright 2015 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/netplugin/core"
+	"github.com/contiv/netplugin/crt"
+)
+
+const (
+	invalidAttr         = ""
+	lldpadContainerName = "lldpad"
+)
+
+// Implement utilities to fetch lldp information aobut connecting neighbors.
+// The lldp information can be used in non cloud environment to automate
+// networking configuration on the physical devices
+
+// LLDPNeighbor lists the neighbor attributes as learned by the host
+type LLDPNeighbor struct {
+	SystemName        string
+	SystemDescription string
+	PortID            string
+	PortDesc          string
+	CiscoACIPodID     string
+	CiscoACINodeID    string
+}
+
+// FetchLLDPInfo reads the lldp information using lldptool
+// A native go lib could be done but it requires that lldpad daemon is
+// accessible natively on the lldp ipc path
+func FetchLLDPInfo(crt *crt.CRT, ifname string) (*LLDPNeighbor, error) {
+	lldp := &LLDPNeighbor{}
+
+	if crt.GetContainerID(lldpadContainerName) == "" {
+		log.Errorf("Error fetching '%s' container information - not sure if it is running", lldpadContainerName)
```

Comment:
I think error should be more certain, so following might be one option. But up to you:

```
Error fetching %q container information - please check if it is running.
```